### DISCUSSION TITLE
Do we still need a Slack workspace?

### DISCUSSION THREAD

**Discussion by myitcv** [cue-lang/cue#1160] [Ideas]
Hi everyone - with the [migration to our new home on GitHub](https://github.com/cue-lang/cue/issues/1078) now well behind us, teething problems almost entirely resolved, it seems a good time to consider what role we want Slack to play going forward. 

As many of you will have seen, I've been trying to push many excellent questions/answers to GitHub Discussions which does a much better job of archiving this material, but also making it more easily discoverable. If we ignore for one second the hiccup of the migration, this appears to have been a success - thank you to everyone who has asked great questions, but especially to those who have helped provide answers.

But this raises the obvious question: **do we need Slack any more?**

Please use emoji to vote:

 :+1: = "yes, we still need Slack!"
 :-1: = "no, we can live without Slack"

Please also reply below with thoughts/ideas etc.

Thank you!

**Comment by seh** [comment]
I expect that some kinds of conversations are still better suited to Slack, such as people advertising or asking for feedback about new CUE-related projects of theirs, or looking to share tips and techniques.

**Reply by myitcv** [reply]
Thanks, @seh. I totally accept and agree with your points. 

The challenge, at least as I see it, is how we capture (or put another way, don't lose) great discussions in Slack. Perhaps the [GitHub integration](https://github.com/integrations/slack) helps in that respect? If so, would be very grateful if someone can share ideas/experience.

**Reply by grantzvolsky** [reply]
@myitcv A possible solution to the discoverability problem is to periodically export the messages and add them as a static asset to the cue/cuelang.org repository. This would allow the discussions to be searched either using github search on the repository or with a search engine via `<query> site:cuelang.org/channels/general`.

**Reply by myitcv** [reply]
Another thought that just came to mind, is that we could use some sort of Slack integration that drops a link to each new GitHub discussion in the `#general` channel. The reason this might be useful is that it's not (AFAIK) possible to only turn on notifications for _only_ discussions in a repository. On the assumption that people are more interested in discussions than bug reports, and furthermore that Slack is a part of many people's notification workflows, perhaps this hits a sweet spot. Because if someone is then interested in a discussion that gets notified to Slack, they can subscribe to that discussion or reply to join in.

Thoughts?

**Reply by verdverm** [reply]
I like a two way syncing.

The latest GitHub workflow has triggers for discussions so it could be automated that way for one direction. Slack to GitHub integration doesn't look to support discussion yet afaik

**Comment by cbrgm** [comment]
I agree with @seh. I think a tool to ask questions directly or to talk directly to community members is important. I also think that for many people Slack is a bit better known than the Github discussions, also for people who are relatively new to the Cue community Slack is a good starting point. 

However, I think another chat tool that is "officially" supported by the community could replace Slack, e.g. Gitter, IRC or the like in the long run, as for me having a direct chat with the community is actually the most important feature.

**Reply by myitcv** [reply]
> However, I think another chat tool that is "officially" supported by the community could replace Slack, e.g. Gitter, IRC or the like in the long run, as for me having a direct chat with the community is actually the most important feature.

Thanks @cbrgm. Just to confirm, we're not set on removing Slack support, neither are we set on using Slack. All options on the table :) But as I just mentioned in my [response to @seh](https://github.com/cue-lang/cue/discussions/1160#discussioncomment-1085371), one of the main challenges is not losing great content to a walled garden. Would other tools help in that respect?

**Comment by nyarly** [comment]
Discussions are _probably_ the better way to handle almost any question about Cue for the near term. The size of the community is such that questions get answered on the hours or next-day cycle, which Discussions would be better for. In the meantime, Slack is a big distraction from Discussions - it's hard to monitor both, and Slack is better at grabbing attention.

**Reply by myitcv** [reply]
Thanks @nyarly. This is definitely a point to consider, but I do appreciate and agree with @seh's point above. Some people find Slack easier, more approachable; at least in my experience of the Go slack channel, it's a valuable medium in that respect. As I said in [my response to @seh](https://github.com/cue-lang/cue/discussions/1160#discussioncomment-1085371), the main challenge as I see it is that in supporting multiple channels how we don't lose good content, or for that matter have content "rot" (i.e. an answer becomes invalid, or no longer best practice).

**Reply by nyarly** [reply]
Rotting content brings to mind a further issue about the relationship between discussion and documentation. I don't think holding on to old conversations is really adequate as documentation, in general. CUE has great docs for the most part, but I know  there are aspects that I've only picked up in conversation. Probably, there's a worthwhile contribution to be made: reading the docs and adding details that seem to be missing.

**Reply by nyarly** [reply]
Separately, there's a quibble to be had about whether Slack is inherently a valuable medium, or if it enables valuable communities. There are certainly projects in the Kubernetes sphere where the Slacks act as living documentation, chiefly because maintainers make a point of being present and answering questions. I share your pleasant experience of the Go Slacks. On the other hand, some are bitbuckets - questions go there to languish, and in part I think that's because there isn't the capacity to reply quickly.

**Comment by verdverm** [comment]
Generally, I think having a chat like capability is valuable as others have mentioned. Slack seems to be where most are with the best support for integrations. Chat is important for building the community and fostering the more casual conversations. It has a lower barrier for engagement and might also be considered the frontline of support.

I am not thrilled with discussion. The search is terrible and cannot even match substrings from a title, requiring you to know the exact and full word to use...

I do like the idea of capturing common questions so they don't need answering, but I still feel like we are doing this on discussions due to the abysmal discoverability within the GH feature.

**Reply by myitcv** [reply]
> Chat is important for building the community and fostering the more casual conversations. It has a lower barrier for engagement and might also be considered the frontline of support.

This point about "chat" is an important one. We're seeing the same pattern with Go, especially modules. Often in complex modules examples it's not clear either what the question should be or what the issue is, and the immediate form of discussion works well for navigating the conversation (although the lack of proper threading sometimes makes multiple conversations hard). 

That said I think it's worth pointing out that Slack isn't the only chat platform; we should consider platforms like Discord to see if they offer better support for threading, retention, moderation etc. This discussion can also be viewed as a means of establishing what we are solving for with chat vs other mediums.  

> The search is terrible and cannot even match substrings from a title, requiring you to know the exact and full word to use...

Have you raised this as feedback with GitHub, or added to an existing discussion/thread on this topic? 

> I do like the idea of capturing common questions so they don't need answering

It's more that we can capture those discussions in a persistent format/medium, which is more easily discoverable.

> but I still feel like we are doing this on discussions due to the abysmal discoverability within the GH feature.

I don't understand this point.
